### DISCUSSION TITLE
0.52 and trailing commas behavior on Kotlinlang formatting

### DISCUSSION THREAD

**Discussion by JavierSegoviaCordoba** [facebookincubator/ktfmt#510] [General]
After upgrading I am seeing trailing commas being removed and I think it is not only worse because it is less flexible in terms of how you can format the code (decide to have multiline or not by adding a trailing comma), but objectively it can modify the code in a way it transforms something from multiline to single-line and vice-versa making the code reviews harder when you want to keep a multiline for readability reason and it becomes a single-line, and in a future when a new parameter is added it becomes again multiline.

```kotlin
data class Foo(
    val a: A,
    val b: B,
)
```

That becomes:

```kotlin
data class Foo(val a: A, val b: B)
```

And if you add more params:

```kotlin
data class Foo(
    val a: A,
    val b: B,
    val c: C,
    val d: D,
    val e: E,
)
```

The format on `constructor` has changed too, personally, I prefer the old one, which follows the "square" pattern:

```diff
- public open class HubdleExtension
- @Inject
- constructor(
-     project: Project,
- ) : HubdleEnableableExtension(project) {

+ public open class HubdleExtension @Inject constructor(project: Project) :
+     HubdleEnableableExtension(project) {
```

**Comment by cortinico** [comment]
@JavierSegoviaCordoba can you open an issue for this one?

**Comment by hick209** [comment]
If you update the trailing comma management style to `only_add` you should get the behavior you are looking for, if I'm not mistaken.
